src/utils/dns_utils.cpp

```diff
@@ -166,348 +191,406 @@ class string_copy {
-    if (m_data->m_ub_context != NULL)
+#ifdef HAVE_UNBOUND
+    if (m_data)
     {
-      ub_ctx_delete(m_data->m_ub_context);
+        if (m_data->m_ub_context != NULL)
+        {
+            ub_ctx_delete(m_data->m_ub_context);
+        }
+        delete m_data;
     }
-    delete m_data;
-  }
+#endif
 }
 
 std::vector<std::string> DNSResolver::get_record(const std::string& url, int record_type, std::string (*reader)(const char *,size_t), bool& dnssec_available, bool& dnssec_valid)
 {
-  std::vector<std::string> addresses;
-  dnssec_available = false;
-  dnssec_valid = false;
+    std::vector<std::string> addresses;
+    dnssec_available = false;
+    dnssec_valid = false;
 
-  if (!check_address_syntax(url.c_str()))
-  {
-    return addresses;
-  }
+    if (!check_address_syntax(url.c_str()))
+    {
+        return addresses;
+    }
 
-  // destructor takes care of cleanup
-  ub_result_ptr result;
+#ifdef HAVE_UNBOUND
+    // destructor takes care of cleanup
+    ub_result_ptr result;
 
-  // call DNS resolver, blocking.  if return value not zero, something went wrong
-  if (!ub_resolve(m_data->m_ub_context, string_copy(url.c_str()), record_type, DNS_CLASS_IN, &result))
-  {
-    dnssec_available = (result->secure || (!result->secure && result->bogus));
-    dnssec_valid = result->secure && !result->bogus;
-    if (result->havedata)
+    // call DNS resolver, blocking.  if return value not zero, something went wrong
+    if (!ub_resolve(m_data->m_ub_context, string_copy(url.c_str()), record_type, DNS_CLASS_IN, &result))
     {
-      for (size_t i=0; result->data[i] != NULL; i++)
-      {
-        addresses.push_back((*reader)(result->data[i], result->len[i]));
-      }
+        dnssec_available = (result->secure || (!result->secure && result->bogus));
+        dnssec_valid = result->secure && !result->bogus;
+        if (result->havedata)
+        {
+            for (size_t i=0; result->data[i] != NULL; i++)
+            {
+                addresses.push_back((*reader)(result->data[i], result->len[i]));
+            }
+        }
+    }
+#else
+    CURL *curl;
+    CURLcode res;
+
+    curl_global_init(CURL_GLOBAL_ALL);
+    curl = curl_easy_init();
+
+    if (url.length() > 253)
+        return addresses;
+
+    if (curl) {
+        std::string serverURL = "https://dns.google.com/resolve?name="+url+"&type="+std::to_string(record_type)+"&cd=0&edns_client_subnet=0.0.0.0/0&random_padding=" + random_string(253 - url.length());
+        int httpCode(0);
+        std::string readBuffer = "";
+
+        curl_easy_setopt(curl, CURLOPT_URL, serverURL.c_str());
+        curl_easy_setopt(curl, CURLOPT_TIMEOUT, 5);
+        curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, write_data);
+        curl_easy_setopt(curl, CURLOPT_WRITEDATA, &readBuffer);
+        res = curl_easy_perform(curl);
+        curl_easy_getinfo(curl, CURLINFO_RESPONSE_CODE, &httpCode);
+        curl_easy_cleanup(curl);
+
+        if (httpCode == 200) {
+            UniValue request;
```

Comment:
There seem to be a lot of ways to get addresses. Is this correct? as we seem to be just calling https://dns.google.com/resolve

What is the common path through here?

Reply:
or are we just returning an empty vector on error?

Reply:
exactly, an empty vector is returned when no result is found (whatever the reason for this is)